express-notes-api/index.js

```diff
@@ -0,0 +1,96 @@
+const fs = require('fs');
+const express = require('express');
+const app = express();
+const add = require('./add');
+const del = require('./delete');
+const update = require('./update');
+const notes = require('./data.json');
+let nextId = 10;
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const notesObject = [];
+  for (const prop in notes) {
+    notesObject.push(notes[prop]);
+  }
+  res.json(notesObject);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (!id) {
+    return res.status(400).send({ error: 'ID must be a positive integer' });
+  }
+  if (!notes.notes[id]) {
+    return res.status(404).send({ error: `Cannot find note with ID ${id}` });
+  }
+
+  res.json(notes.notes[id]);
+
+});
+
+app.post('/api/notes', (req, res) => {
+  const data = req.body;
+  if (!req.is('application/json') || req.is() === null) {
```

Comment:
This will allow any property in the body to be used for a note, you want to directly check if there is a `content` property on the `body`. Also it can be simplified by using something like:

```JAVASCRIPT
if (!data.content) {
  // No content on body, send error
}
```